isis/tests/Fixtures.cpp

```diff
@@ -333,89 +302,19 @@ namespace Isis {
   }
 
   void ApolloNetwork::TearDown() {
-    if (cube1->isOpen()) {
-      cube1->close();
-    }
-
-    if (cube2->isOpen()) {
-      cube1->close();
-    }
-
-    if (cube3->isOpen()) {
-      cube1->close();
-    }
-
-    if (cube4->isOpen()) {
-      cube1->close();
-    }
-
-    if (cube5->isOpen()) {
-      cube1->close();
-    }
-
-    if (cube6->isOpen()) {
-      cube1->close();
-    }
-
-    if (cube7->isOpen()) {
-      cube1->close();
-    }
-
-    if (isdFile1) {
-      delete isdFile1; 
-    }
-
-    if (isdFile2) {
-      delete isdFile2;
-    }
-
-    if (isdFile3) {
-      delete isdFile3; 
-    }
-
-    if (isdFile4) {
-      delete isdFile4; 
-    }
-
-    if (isdFile5) {
-      delete isdFile5; 
-    }
-
-    if (isdFile6) {
-      delete isdFile6; 
-    }
-
-    if (isdFile7) {
-      delete isdFile7; 
-    }
-
-    if (cube1) {
-      delete cube1; 
-    }
-
-    if (cube2) {
-      delete cube2; 
-    }
-
-    if (cube3) {
-      delete cube3; 
-    }
-
-    if (cube4) {
-      delete cube4; 
-    }
-
-    if (cube5) {
-      delete cube5; 
-    }
+    for(int i = 0; i < cubes.size(); i++) {
+      if(cubes[i] && cubes[i]->isOpen()) {
+        cubes[i]->close();
```

Comment:
You should just delete cubes[i] if it's non-null. The destructor will close the cube for you.